scripts/core/editor3/helpers/highlights.js

```diff
@@ -845,6 +844,24 @@ export function getRangeAndTextForStyle(editorState, style) {
     };
 }
 
+export function getRangeAndTextForStyleInRawState(rawEditorState, highlightId) {
```

Comment:
I tried to use an actual editor state by creating one from scratch based on the raw state. It didn't work because our functions try to get the selection (which doesn't exist on the created state). Otherwise I would use the one we already have (`getRangeAndTextForStyle`)